src/Builder/Creator/MethodStatementsCreator.php

```diff
@@ -0,0 +1,165 @@
+    {
+        if (is_array($value)) {
+            return $this->createNodeTreeFromArrayValue($classInformationTransfer, $value);
+        }
+
+        return $this->createNodeTreeFromStringValue($classInformationTransfer, $value);
+    }
+
+    /**
+     * @param \SprykerSdk\Integrator\Transfer\ClassInformationTransfer $classInformationTransfer
+     * @param mixed $value
+     *
+     * @return array
+     */
+    protected function createNodeTreeFromStringValue(ClassInformationTransfer $classInformationTransfer, $value): array
+    {
+        $arrayItems = [];
+        $valueItems = explode(
+            '::',
+            $this->getShortClassNameAndAddToClassInformation($classInformationTransfer, $value),
+        );
+        $arrayItems[] = new ArrayItem(
+            $this->createClassConstantExpression($valueItems[0], $valueItems[1]),
+        );
+
+        return $arrayItems;
+    }
+
+    /**
+     * @param \SprykerSdk\Integrator\Transfer\ClassInformationTransfer $classInformationTransfer
+     * @param mixed $value
+     *
+     * @return array
+     */
+    protected function createNodeTreeFromArrayValue(ClassInformationTransfer $classInformationTransfer, $value): array
+    {
+        $arrayItems = [];
+        foreach ($value as $key => $item) {
+            $insideArrayItems = [];
+            $itemParts = [];
+
+            if (is_string($item)) {
+                $item = $this->getShortClassNameAndAddToClassInformation($classInformationTransfer, $item);
+                $itemParts = explode('::', $item);
+            } else {
+                $insideArrayItems = $this->createMethodStatementsFromValue($classInformationTransfer, $item);
+            }
+
+            if (is_int($key)) {
+                $arrayItems[] = $this->createArrayItem($itemParts);
+
+                continue;
+            }
+
+            $key = $this->getShortClassNameAndAddToClassInformation($classInformationTransfer, $key);
+            $keyParts = explode('::', $key);
+            if (is_array($item)) {
+                $arrayItems[] = $this->createArrayItem([], $keyParts, $insideArrayItems);
+
+                continue;
+            }
+
+            $arrayItems[] = $this->createArrayItem($itemParts, $keyParts);
+        }
+
+        return $arrayItems;
+    }
+
+    /**
+     * @param array $itemParts
+     * @param array $keyParts
+     * @param array $insideArrayItems
+     *
+     * @return \PhpParser\Node\Expr\ArrayItem
+     */
+    protected function createArrayItem(array $itemParts, array $keyParts = [], array $insideArrayItems = []): ArrayItem
+    {
+        if (count($itemParts) === static::SIMPLE_VARIABLE_SEMICOLON_COUNT) {
+            return $this->createSingleSemicolonVariableArrayItem($itemParts, $keyParts);
+        }
+
+        if ($insideArrayItems) {
+            return new ArrayItem(
+                (new BuilderFactory())->val($insideArrayItems),
+                $this->createClassConstantExpression($keyParts[0], $keyParts[1]),
```

Comment:
would be nice to have constants for this keys with descriptive name

Reply:
What do you mean?

Reply:
I mean magic numbers 0 and 1. What values do we get from those? What do we have in array element with index 0 and what do we have for array element with index 1?